akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala

```diff
@@ -648,9 +648,15 @@ class ClusterSingletonManager(singletonProps: Props, terminationMessage: Any, se
       stay.using(YoungerData(None))
 
     case Event(HandOverToMe, _) =>
-      // this node was probably quickly restarted with same hostname:port,
-      // confirm that the old singleton instance has been stopped
-      sender() ! HandOverDone
+      val selfStatus = cluster.selfMember.status
+      if (selfStatus == MemberStatus.Leaving || selfStatus == MemberStatus.Exiting)
+        logInfo("Ignoring HandOverToMe in Younger from [{}] because self is [{}].", sender().path.address, selfStatus)
```

Comment:
Hmm, so we are on the previous oldest, because we are leaving, and got a request to hand over from another node, won't this always be true? 

Isn't the problem about the status of the node requesting the handover when `HandOverDone` arrives?

Reply:
Sorry, read the explanation above now. 👍

Reply:
No, the purpose of this is as commented a few lines down:
```
        // this node was probably quickly restarted with same hostname:port,
        // confirm that the old singleton instance has been stopped
```

This is in state Younger.

The way it was triggered for the failure scenario was that node-1 was the singleton, and then leaving node-1 and node-2. Then node3 could see node-2 as exiting and therefore believe that node-2 was previous oldest instead of node-3. But, node-2 itself had not seen the exiting yet so it was still in Younger and confirmed the request from node-3 although the real singleton actor was still running on node-1.

Reply:
I'm wondering if this is the issue I'd seen before, where there were > 1 instances. Would be fantastic if it is, and this prevents that.